int8 vertex1
int8 vertex2
float32 distance